jdatetime/__init__.py

```diff
@@ -531,131 +562,67 @@ def __format__(self, format):
         """
         return self.strftime(format)
 
-    def strftime(self, format):
-        """format -> strftime() style string."""
-        # TODO: change stupid str.replace
-        # formats = {
-        #           '%a': lambda: self.j_weekdays_short[self.weekday()]
-        # }
-        # find all %[a-zA-Z] and call method if it in formats
-
-        # convert to unicode
-        try:
-            format = format.decode('utf-8')
-        except Exception:
-            pass
-
-        format = format.replace("%a", self.j_weekdays_short[self.weekday()])
+    def _strftime_replace_func(self, symbol, value, fallback=None):
+        def simple_replace(f):
+            return f.replace(symbol, value(self))
 
-        format = format.replace("%A", self.j_weekdays[self.weekday()])
-
-        format = format.replace("%b", self.j_months_short[self.month - 1])
-
-        format = format.replace("%B", self.j_months[self.month - 1])
-
-        if '%c' in format:
-            format = format.replace("%c", self.strftime("%a %b %d %H:%M:%S %Y"))
-
-        format = format.replace("%d", '%02.d' % (self.day))
-        format = format.replace("%-d", '%d' % (self.day))
-
-        try:
-            format = format.replace("%f", '%06.d' % (self.microsecond))
-        except Exception:
-            format = format.replace("%f", "000000")
-
-        try:
-            format = format.replace("%H", '%02.d' % (self.hour))
-        except Exception:
-            format = format.replace("%H", '00')
-
-        try:
-            format = format.replace("%-H", '%d' % (self.hour))
-        except Exception:
-            format = format.replace("%-H", '0')
-
-        try:
-            format = format.replace("%I", '%02.d' % (self.hour % 12 or 12))
-        except Exception:
-            format = format.replace("%I", '12')
+        def replace_with_fallback(f):
+            try:
+                return f.replace(symbol, value(self))
+            except Exception:
```

Comment:
maybe catching a limited list of exceptions is a better option? say `ValueError` and `TypeError`?

Reply:
Yes, it's better to catch just a limited set of exceptions. but it was like that [in the previous implementation](https://github.com/slashmili/python-jalali/blob/main/jdatetime/__init__.py#L567-L570).
we don't have enough tests to cover these exceptions. I would suggest to keep it like that to don't break something.